lib/macro/macro-action-variable.cpp

```diff
@@ -906,21 +931,21 @@ void MacroActionVariableEdit::UpdateEntryData()
 	_actions->setCurrentIndex(
 		_actions->findData(static_cast<int>(_entryData->_type)));
 	_strValue->setPlainText(_entryData->_strValue);
-	_numValue->setValue(_entryData->_numValue);
+	_numValue->SetValue(_entryData->_numValue);
 	_segmentIdx->SetValue(_entryData->GetSegmentIndexValue() + 1);
 	_segmentIdx->SetMacro(_entryData->GetMacro());
 	_segmentIdx->SetType(
 		_entryData->_type ==
 				MacroActionVariable::Type::SET_CONDITION_VALUE
 			? MacroSegmentSelection::Type::CONDITION
 			: MacroSegmentSelection::Type::ACTION);
-	_subStringStart->setValue(_entryData->_subStringStart + 1);
-	_subStringSize->setValue(_entryData->_subStringSize);
+	_subStringStart->SetFixedValue(_entryData->_subStringStart + 1);
```

Comment:
`SetValue()` needs to be used here if you want the variable selection to respected as well.
Otherwise the selection will always switch back to the non-variable selection.

Reply:
Ah. There were some issues with implicit conversion here, but I'll recheck

Reply:
Don't know why it was done so weirdly in the 1st place, but should be fine now